### PR TITLE
Add chapter about Applying Core patches

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: Documentation/ApplyingCorePatches/Index.rst [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+   Composer. If you still run a legacy TYPO3 installation consider to
+   :ref:`migrate to Composer <migratetocomposer>`.
+
+To automatically apply patches first install `cweagans/composer-patches`:
+
+.. code-block:: bash
+
+   composer req cweagans/composer-patches
+
+Chose a folder to store all patches in. This folder should ideally lie outside
```
Comment (brotkrueml) [inline Documentation/ApplyingCorePatches/Index.rst:41]:
```suggestion
Choose a folder to store all patches in. This folder should ideally lie outside
```

File: Documentation/ApplyingCorePatches/Index.rst [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+   Could not apply patch! Skipping. The error was: Cannot apply patch patches/Bug-98106.diff
+
+You can get a more verbose error message by calling:
+
+.. code-block:: bash
+   :caption: Very, very verbose error message
+
+   composer install -vvv
+
+Creating a diff from a Core change
```
Comment (brotkrueml) [inline Documentation/ApplyingCorePatches/Index.rst:118]: We should also mention https://packagist.org/packages/gilbertsoft/typo3-core-patches in this chapter, as this greatly simplifies applying core patches.

Comment (linawolf) [reply Documentation/ApplyingCorePatches/Index.rst:118]: I never used it. Would you like to add a chapter, @brotkrueml ?

Comment (brotkrueml) [reply Documentation/ApplyingCorePatches/Index.rst:118]: Done

**Comment by brotkrueml** [comment]
Should we backport it to 11.5?

**Comment by sypets** [comment]
> Should we backport it to 11.5?

Good idea.

File: Documentation/ApplyingCorePatches/Index.rst [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+   Composer. If you still run a legacy TYPO3 installation consider to
+   :ref:`migrate to Composer <migratetocomposer>`.
+
+To automatically apply patches first install `cweagans/composer-patches`:
+
+.. code-block:: bash
+
+   composer req cweagans/composer-patches
+
+Choose a folder to store all patches in. This folder should ideally lie outside
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:41]:
```suggestion
Choose a folder to store all patches in. This folder should ideally be outside
```

File: Documentation/ApplyingCorePatches/Index.rst [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+of the webroot. Here we use the folder :file:`patches` on the same level like
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:42]:
```suggestion
of the webroot. Here we use the folder :file:`patches` on the same level as
```

File: Documentation/ApplyingCorePatches/Index.rst [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+
+
+Apply the patch by running the following command:
+
+.. code-block:: bash
+
+   composer install
+
+.. note::
+   Changes to the patch file after the patch was applied successfully once will
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:100]:
```suggestion
   Changes to the patch file after the patch was applied successfully will
```

I think the "once" is unnecessary and it tripped me up on reading.

File: Documentation/ApplyingCorePatches/Index.rst [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+   not be automatically applied. In that case delete the installed sources
+   then execute :bash:`composer install` once more.
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:102]:
```suggestion
   and execute :bash:`composer install` once more.
```

File: Documentation/ApplyingCorePatches/Index.rst [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+
+If applying the patch fails you will often get a cryptic error message like:
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:104]:
```suggestion
If applying the patch fails, you may get a cryptic error message like:
```

File: Documentation/ApplyingCorePatches/Index.rst [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+
+-  :ref:`Apply a core patch manually <apply-core-patch-manually>`
+-  :ref:`Apply a core patch automatically via gilbertsoft/typo3-core-patches <apply-core-patch-automatically>`
+
+.. _apply-core-patch-manually:
+
+Apply a core patch manually
+---------------------------
+
+In case a new Core version has not been released yet, but your urgently need
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:131]:
```suggestion
In case a new Core version has not been released yet, but you urgently need
```

**Comment by sypets** [comment]
I am reviewing, please do not merge yet.

File: Documentation/ApplyingCorePatches/Index.rst [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+
+Then look up the change ID on `review.typo3.org <https://review.typo3.org/>`.
+You can find it in the URL or left of the title of the change. In the example
+it is `75368`.
+
+.. figure:: /Images/ManualScreenshots/ReviewChangeId.png
+   :class: with-shadow
+   :alt: Look up of the change id
+
+   Look up of the change id
```
Comment (sypets) [inline Documentation/ApplyingCorePatches/Index.rst:262]:
```suggestion
   Look up the change id
```

File: Documentation/ApplyingCorePatches/Index.rst [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+
+   composer req gilbertsoft/typo3-core-patches
+
+Then look up the change ID on `review.typo3.org <https://review.typo3.org/>`.
+You can find it in the URL or left of the title of the change. In the example
+it is `75368`.
+
+.. figure:: /Images/ManualScreenshots/ReviewChangeId.png
+   :class: with-shadow
+   :alt: Look up of the change id
```
Comment (brotkrueml) [inline Documentation/ApplyingCorePatches/Index.rst:260]:
```suggestion
   :alt: Look up the change id
```

**Review by sypets — APPROVED** [review]

**Review by brotkrueml — APPROVED** [review]

File: Documentation/ApplyingCorePatches/Index.rst [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+                if ($mode === 'embed') {
+                    $mode = true;
+
+
+.. _apply-core-patch-automatically:
+
+Apply a core patch automatically via `gilbertsoft/typo3-core-patches`
+---------------------------------------------------------------------
+
+With the help of the Composer package `gilbertsoft/typo3-core-patches` a core
```
Comment (linawolf) [inline Documentation/ApplyingCorePatches/Index.rst:243]:
```suggestion
With the help of the Composer package `gilbertsoft/typo3-core-patches` a Core
```

**Review by tomwarwick — APPROVED** [review]

**Pull request merged**